posit-dev/positron
# Bundle Pandoc with Positron

### Intent

This change causes us to start bundling Pandoc as part of Positron, so that it can be available to various utilities that depend on it. Pandoc is used heavily by tools that use R Markdown directly or indirectly, such as building package vignettes, reprex, etc. 

Addresses https://github.com/posit-dev/positron/issues/1466.

### Approach

- Download a version-pinned Pandoc binary appropriate for the current OS during the packaging step.
- Bundle the Pandoc binary in the `bin` folder appropriate for the current us. 
- In the R language pack, set the environment variable `RSTUDIO_PANDOC` to the location of this folder. Most R-based tools check this environment variable to for a bundled copy of Pandoc.

### QA Notes

- This deserves testing across all operating systems since each has its own path/naming quirks.
- Remove any copies of `pandoc` from your `$PATH` when testing to ensure R uses the bundled copy.
- I recommend using `rmarkdown::find_pandoc()` as a test to make sure R is picking up what we're putting down.
